frontend/src/components/software/SoftwareDetailsContainer.jsx

```diff
@@ -2,9 +2,60 @@ import React, {Component} from 'react';
 import SoftwareDetailsComponent from './SoftwareDetailsComponent';
 
 class SoftwareDetailsContainer extends Component {
+  constructor(props) {
+    super(props);
+    this.addCall = this.addCall.bind(this);
+    this.state = {
+      name: '',
+      key: '',
+      availableKeys: '',
+      duration: '',
+      mode: '',
+      id: '',
+    };
+  }
+
+
+
+  addCall(){
+    fetch('http://localhost:8080/api/software',{
+      method: 'POST',
+      mode: 'cors',
+      body: JSON.stringify(this.state),
+      headers: {
+        'Content-Type': 'application/json',
+        'Access-Control-Allow-Origin': '*',
+      }
+    }).then((response) => response.json())
+      .then((responseJson) => {
+        console.log(responseJson);
+        return responseJson;
+      })
+      .catch((error) => {
+        console.error(error);
+      });
+  }
+
+  onSubmit = () => this.addCall();
+  onReject = () => this.props.goBack();
+  setName = (value) => this.setState({name: value});
+  setKey = (value) => this.setState( {key: value});
+  setAvailableKeys = (value) => this.setState({availableKeys: value});
+  setDuration = (value) => this.setState({duration: value});
+  getState() {return this.state.mode};
   render() {
+    this.state.mode = this.props.mode;
```

Comment:
Niedozwolone jest bezpośrednie ustawianie `this.state`, gdyż powoduje to trudne do zdebugowania błędy. `this.state = ` może być użyte tylko w konstruktorze, wszędzie indziej należy używać `setState`. Metoda `setState` nie może być użyta wewnątrz funkcji `render`.

Reply:
Tak, jak napisałem powyżej - w ogóle nie musimy tutaj mieć tej linijki, bo zawsze mamy dostępne `this.props`.

Reply:
poprawione